src/mail.sh

```diff
@@ -0,0 +1,206 @@
+}
+
+# This function adds the supplied configuration option. It first checks if the
+# given option has been set, then prompts the user to replace, add, or ignore
+# the configuration
+#
+# @option:     The full option being edited
+# @value:      The value to update the option to
+# @set_values: The values of the option that are already set
+#
+# Return:
+# Returns 0 if successful; non-zero otherwise
+function add_config()
+{
+  local flag="$1"
+  local option="$2"
+  local setting="${options_values["$option"]% \'*}" # the setting to be set
+  local value="${options_values["$option"]#* }"     # the value to be set
+  local curr_scope="${options_values['GLOBAL']#--}"
+  local -a set_values
+  local cmd=''
+  local warn=0
+  local i=0
+
+  flag=${flag:-'SILENT'}
+  curr_scope=${curr_scope:-'local'}
+
+  # Gets set configurations and their scope
+  # The parameter '-d' is used to read all lines at once
+  IFS=$'\t\n' read -r -a set_values -d '' <<< "$(get_all_git_config "$flag" "$setting")"
+
+  if [[ "${#set_values}" -gt 0 ]] && [[ "${set_values[*]}" =~ $curr_scope ]]; then
+    warning "The configuration $setting is already set with the following value(s):"
+    while [[ -n "${set_values["$i"]}" ]]; do
+      warning "  '${set_values["$((i + 1))"]}' at a ${set_values["$i"]} scope"
+      if [[ "${set_values["$i"]}" == "$curr_scope" ]]; then
+        warn=1
+      fi
+      i=$((i + 2))
+    done
+
+    if [[ "$warn" == 1 ]]; then
+      printf '%s\n' '' "You are currently editing at a $curr_scope scope." \
+        'If you continue the value at this scope will be overwritten.' \
+        "The new value will be $value"
+
+      if [[ $(ask_yN 'Do you wish to proceed?') == 0 ]]; then
+        complain "Skipping $setting..."
+        return 1
+      fi
+    fi
+  fi
+
+  cmd="git config ${options_values['GLOBAL']} $setting $value"
+  cmd="$(str_remove_duplicates "$cmd" ' ')"
+
+  cmd_manager "$flag" "$cmd"
+}
+
+function parse_mail_options()
+{
+  local short_options='h,t,'
+
+  local long_options='help,setup,global,'
+  long_options+='email:,name:,'
+  long_options+='smtpuser:,smtpencryption:,smtpserver:,smtpserverport:,smtppass:,'
+
+  options="$(kw_parse "$short_options" "$long_options" "$@")"
+  if [[ "$?" != 0 ]]; then
+    options_values['ERROR']="$(kw_parse_get_errors 'kw mail' "$short_options" \
+      "$long_options" "$@")"
+    return 22 # EINVAL
+  fi
+
+  # Default values
+  options_values['SETUP']=0
+  options_values['GLOBAL']=''
+
+  eval "set -- $options"
+
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      --help | -h)
+        mail_help "$1"
+        exit
+        ;;
+      --setup | -t)
+        options_values['SETUP']=1
+        shift
+        ;;
+      --email | --name)
+        local index
+        local option="${1#--}"
+        index="$(str_uppercase "$option")"
+        options_values["$index"]="user.$option '$2'"
+        shift 2
+        ;;
+      --smtpencryption)
+        validate_encryption "$2"
```

Comment:
There is no reason that the rest of the configuration can't continue even with a bad encryption, so there is no need to halt execution. But what do you suggest we do in case of failure?